### PR TITLE
Improve docs of ILogger

### PR DESCRIPTION
**Pull request by 304NotModified**
Supersedes https://github.com/aspnet/Logging/pull/342

Not sure what the requirements are of the docs (not found in [CONTRIBUTING.md](https://github.com/aspnet/Home/blob/dev/CONTRIBUTING.md)), but missing docs could be a source of bugs.

### CODE REVIEW

**Comment by Eilon** [comment]
@kichalla can you review?

**Comment by Eilon** [comment]
@kichalla bump.

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs [hunk 1]
```diff
@@ -8,24 +8,24 @@ namespace Microsoft.Extensions.Logging
     /// <summary>
     /// Represents a type used to perform logging.
     /// </summary>
+    /// <remarks>Aggregates most logging patterns to a single method.</remarks>
     public interface ILogger
     {
         /// <summary>
-        /// Aggregates most logging patterns to a single method.
+        /// Writes an event to the logger.
```
Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: Not sure if `event` is the right word to use here. Can it just be `Writes a log entry`? @Eilon

Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: :+1:

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs [hunk 2]
```diff
@@ -8,24 +8,24 @@ namespace Microsoft.Extensions.Logging
...
         /// </summary>
-        /// <param name="logLevel"></param>
-        /// <param name="eventId"></param>
-        /// <param name="state"></param>
-        /// <param name="exception"></param>
-        /// <param name="formatter"></param>
-        /// <returns></returns>
+        /// <param name="logLevel">Event will be written on this level.</param>
```
Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: Remove the references to `event` in the below docs also.

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs [hunk 3]
```diff
@@ -8,24 +8,24 @@ namespace Microsoft.Extensions.Logging
...
+        /// <param name="eventId">Id of this event.</param>
```
Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: This is fine to stay

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs (hunk 1 above)
Comment (304NotModified) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: I like the 'entry' also better, but there is a parameter 'eventId' ? (and not entryId)

**Comment by 304NotModified** [comment]
I have updated the PR.

If it's all OK, I can squash them before merge.

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs [hunk 4]
```diff
@@ -8,24 +8,24 @@ namespace Microsoft.Extensions.Logging
     /// <summary>
     /// Represents a type used to perform logging.
     /// </summary>
+    /// <remarks>Aggregates most logging patterns to a single method.</remarks>
     public interface ILogger
     {
         /// <summary>
-        /// Aggregates most logging patterns to a single method.
+        /// Writes a log entry to the logger.
```
Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: remove `to the logger`

File: src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs [hunk 5]
```diff
@@ -8,24 +8,24 @@ namespace Microsoft.Extensions.Logging
...
-        /// <param name="logLevel"></param>
-        /// <param name="eventId"></param>
-        /// <param name="state"></param>
-        /// <param name="exception"></param>
-        /// <param name="formatter"></param>
-        /// <returns></returns>
+        /// <param name="logLevel">Entry will be written on this level.</param>
+        /// <param name="eventId">Id of this event.</param>
+        /// <param name="state">The entry to be written. Can be also an object.</param>
+        /// <param name="exception">The exception correlated to this entry.</param>
```
Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Abstractions/ILogger.cs]: `correlated` => `related`?

**Comment by 304NotModified** [comment]
Updated after review.

**Comment by kichalla** [comment]
:shipit:

**Comment by 304NotModified** [comment]
OK, let me know if can squash it.

**Comment by kichalla** [comment]
Sure. Please squash and rebase it.

**Comment by 304NotModified** [comment]
Squash done! :)

**Comment by 304NotModified** [comment]
@kichalla bump

**Comment by kichalla** [comment]
c0ccd360ebdd38d132881666e016c163926ad97d Thanks!

**Pull request closed without merging**